core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/worker/DataProcessor.scala

```diff
@@ -154,7 +158,11 @@ class DataProcessor(
       case FinalizePort(portId, input) =>
         asyncRPCClient.send(PortCompleted(portId, input), CONTROLLER)
       case schemaEnforceable: SchemaEnforceable =>
-        statisticsManager.increaseOutputTupleCount()
+        if (outputPortOpt.isEmpty && !outputManager.getSingleOutputPortIdentity.internal) {
```

Comment:
same here, it's better to collect internal ports but not report them outside.

Reply:
I'll fix this in the different PR